### PR TITLE
Audits usage of isanimal() vs isanimal_or_basic_mob()

### LINKED ISSUE #68881: Leaving anomolous crystal posession just gibs you
## Reproduction:

Use the critter posession crystal next to Ian, then press the "leave possesion" HUD icon. Congratulation you're now gibbed

This actually sucked since I was in the most perfect opportunity to assasinate the unsuspecting HoP, but I just ended up covering him in my gibs 
image

### PR DESCRIPTION
**Pull request by Jacquerel**
## About The Pull Request

There's a couple of open issues which fix places where only simple animals were considered, but they are doing it piecemeal.
I decided to just go through every instance of `isanimal` or `subtypesof(mob/living/simple_animal)` I could find, identify which should also affect basic mobs, and fix them.

I left out the two others which are already in PR, I'm not stealing your GBP.

Fixes https://github.com/tgstation/tgstation/issues/68881

## Why It's Good For The Game

Consistency, mostly. 
As far as I can tell all of these things _should_ have effected basic mobs, but didn't.
This fixes a fair number of bugs but also they're bugs that nobody noticed or reported.

There are a couple of places I did not update which will need updating in future. These are:
- Dextrousness checks, because basic mobs don't have that yet.
- The Charge cooldown action, because frankly I couldn't tell what it was trying to do.

alright here goes

## Changelog

:cl:
fix: Carp will once again be healed from being near carp rifts
fix: Sepia slime cores and the rewind camera now work on Ian
fix: Sapient ridden carp (or cows) can throw off their riders by shoving them, or by performing the spin emote.
fix: Giant Spider AI will be disabled by the timestop spell
fix: Ian can eat envirochow
fix: Mice, Frogs, and Cockroaches will no longer set off bear traps
fix: You can put a macrobomb implant into Cayenne (or Ian)
fix: Ian will now recognise that being squeezed by a cyborg is a nice hug
fix: The player panel will tell admins if you're currently a corgi
fix: The staff of storms deals massive damage to Bileworms and Giant Spiders
fix: Ian will whimper if forced to scream
fix: Slimes can consume space carp
fix: Mice can be captured in xenoballs
fix: You can use pacifying potions on Giant Spiders
fix: Sgt Araneus can be fitted with a xenobiological radio implant
fix: Sapient corgis no longer count as living players for the purpose of highlander escape objectives
fix: The random sentience event can now target corgis and sergeant araneus
add: The random sentience event can target a wider array of farm animals
fix: Petsplosion wizard event can target corgis
add: Petsplosion wizard event will now target farm animals and mothroaches
fix: The colossus possession crystal can now actually possess the cockroach it spawns, does not kill you instantly upon ending possession
/:cl:

### CODE REVIEW

File: code/modules/mob/living/simple_animal/hostile/megafauna/colossus.dm
Comment (Sealed101) [inline code/modules/mob/living/simple_animal/hostile/megafauna/colossus.dm:1]: while it isn't using `isanimal()`, it is basically the same issue, preventing the possession of basic mobs

sidenot: that whole method of possession is just hilariously cursed in all the fun ways

https://github.com/tgstation/tgstation/blob/413f2413144e658b8b8f687d3834a265c84db05c/code/modules/mob/living/simple_animal/hostile/megafauna/colossus.dm#L546-L558

Comment (Jacquerel) [reply code/modules/mob/living/simple_animal/hostile/megafauna/colossus.dm:1]: This seems to apply to the random sapience event too.
I should have also searched `for(var/mob/living/simple_animal`...

**Comment by Sealed101** [comment]
I'm guessing it fixes #68881 now

**Comment by Jacquerel** [comment]
> I'm guessing it fixes #68881 now

Yes, although frankly I am also writing down to maybe come and do a bigger rewrite later because it's still fairly stinky
And this probably applies to other anomalous crystal functions

**Comment by san7890** [comment]
i merged a few of the changes i assume you were mentioning in the PR body and no conflicts showed up, which is a bit spooky? not sure, but can you merge master so i can sleep at night

**Pull request merged**